### PR TITLE
fix: improve journal implementation and tests

### PR DESCRIPTION
**Pull request by pvlugter**
Follow up to #2. Changes from reviewing and trying things out locally.

### CODE REVIEW

**Comment by pvlugter** [comment]
Only failing on the deletion related tests now, as expected.

File: core/src/main/scala/akka/persistence/dynamodb/internal/JournalDao.scala [hunk 1]
```diff
@@ -121,4 +122,29 @@ import software.amazon.awssdk.services.dynamodb.model.TransactWriteItemsRequest
...
+    import JournalAttributes._
+
+    val attributeValues = Map(":pid" -> AttributeValue.fromS(persistenceId)).asJava
+
+    val request = QueryRequest.builder
+      .tableName(settings.journalTable)
+      .keyConditionExpression(s"$Pid = :pid")
+      .expressionAttributeValues(attributeValues)
+      .scanIndexForward(false) // get last item (highest sequence nr)
+      .limit(1)
```
Comment (patriknw) [inline core/src/main/scala/akka/persistence/dynamodb/internal/JournalDao.scala:135]: good

File: core/src/test/scala/akka/persistence/dynamodb/TestDbLifecycle.scala [hunk 2]
```diff
@@ -89,6 +90,11 @@ trait TestDbLifecycle extends BeforeAndAfterAll { this: Suite =>
     existingTable.transformWith {
       case Success(_)                            => delete().flatMap(_ => create())
       case Failure(_: ResourceNotFoundException) => create()
+      case Failure(exception: CompletionException) =>
+        exception.getCause match {
+          case _: ResourceNotFoundException => create()
+          case failure                      => Future.failed[Done](failure)
```
Comment (patriknw) [inline core/src/test/scala/akka/persistence/dynamodb/TestDbLifecycle.scala:96]: ah, I probably ran it locally without delete-create and then had the table in place

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**